### ISSUE TITLE
shared library error when using indoc

### ISSUE THREAD

**Issue by glennpierce** [dtolnay/indoc#15]
I am using a nightly compiler and when I use indow I get the error

./target/debug/lasso: error while loading shared libraries: libstd-9f780c2031976271.so: cannot open shared object file: No such file or directory

When running my executable directly.

**Comment by dtolnay** [comment]
I have not been able to reproduce this yet. Could you provide some more details? Are you using indoc from a no_std crate? Is your crate a library or binary? What is the date of the nightly you are using? Does it work with the stable compiler? What platform are you on? Do other procedural macro crates work? Can you share a minimal piece of code that reproduces this?

**Comment by glennpierce** [comment]
So I have a library I developed and some tests that use this library. 

In my Cargo.toml I have included indoc = "0.2"
That's works.

when I include the lib in lib.rs with

#[macro_use]
extern crate indoc;

I get 

./target/release/lasso: error while loading shared libraries: libstd-9f780c2031976271.so: cannot open shared object file: No such file or directory


My compiler is

nightly-x86_64-unknown-linux-gnu
rustc 1.21.0-nightly (b75d1f0ce 2017-08-02)


Trying with the crate quote = "0.3" things seem to work.

I can't try stable as I depend on cron crate that does not work on stable. If you still can't replicate I can start trying to remove code.

Thanks

**Comment by dtolnay** [comment]
Also on nightly-x86_64-unknown-linux-gnu, this script succeeds for me. Any idea what is different in your setup?

```sh
#!/bin/bash

cargo new repro
cat >> repro/Cargo.toml <<-'EOF'
    indoc = "0.2"
EOF

cat > repro/src/lib.rs <<-'EOF'
    #[macro_use]
    extern crate indoc;
    
    pub fn f() -> &'static str {
        indoc!("1
                2")
    }
    
    #[test]
    fn test_indoc() {
        assert_eq!(f(), "1\n2");
    }
EOF

rustc +nightly-2017-08-03 --version
cargo +nightly-2017-08-03 build --manifest-path repro/Cargo.toml
cargo +nightly-2017-08-03 test --manifest-path repro/Cargo.toml
```

**Comment by glennpierce** [comment]
Ok

Change Cargo.toml to

[package]
name = "repro"
version = "0.1.0"
authors = ["Glenn Pierce <glennpierce@gmail.com>"]

[dependencies]
    indoc = "0.2"

[lib]
name = "libtest"
path = "src/lib.rs"
test = false
bench = false

[[bin]]
name = "test"
path = "test/test.rs"


test/test.rs

extern crate libtest;

fn main() {

}

cargo build --release --bin test
./target/release/test 
./target/release/test: error while loading shared libraries: libstd-8eeda694d27cc3ba.so: cannot open shared object file: No such file or directory

**Comment by glennpierce** [comment]
I also tested this on stable with the same result

**Comment by dtolnay** [comment]
Thanks, I was able to reproduce this. I reported it in https://github.com/rust-lang/cargo/issues/4651 as a Cargo bug.

**Comment by dtolnay** [comment]
I implemented a workaround in ac5c0691184e02b5348095baa8ed0a6de38ace84. Could you confirm that you are able to run binaries depending on indoc 0.2.2?

**Comment by glennpierce** [comment]
Yep that seems like it is fixed. Thanks
